examples/lock-app/cyw30739/src/LockManager.cpp

```diff
@@ -275,58 +300,77 @@ bool LockManager::SetUser(uint16_t userIndex, chip::FabricIndex creator, chip::F
 
     for (size_t i = 0; i < totalCredentials; ++i)
     {
-        mCredentials[i]                 = credentials[i];
-        mCredentials[i].CredentialType  = 1;
-        mCredentials[i].CredentialIndex = i + 1;
+        mCredentials[adjustedUserIndex][i]                 = credentials[i];
+        mCredentials[adjustedUserIndex][i].CredentialType  = 1;
+        mCredentials[adjustedUserIndex][i].CredentialIndex = i + 1;
     }
 
-    userInStorage.credentials = chip::Span<const DlCredential>(mCredentials.Get(), totalCredentials);
+    userInStorage.credentials = chip::Span<const DlCredential>(mCredentials[adjustedUserIndex].Get(), totalCredentials);
 
     // Save user information in NVM flash
-    CYW30739Config::WriteConfigValueBin(CYW30739Config::kConfigKey_LockUser, reinterpret_cast<const uint8_t *>(&userInStorage),
-                                        sizeof(EmberAfPluginDoorLockUserInfo));
+    CYW30739Config::WriteConfigValueBin(CYW30739Config::kConfigKey_LockUser, reinterpret_cast<const uint8_t *>(&mLockUsers),
+                                     sizeof(EmberAfPluginDoorLockUserInfo) * mMaxUsers);
 
     CYW30739Config::WriteConfigValueBin(CYW30739Config::kConfigKey_UserCredentials,
-                                        reinterpret_cast<const uint8_t *>(mCredentials.Get()), sizeof(DlCredential));
+                                     reinterpret_cast<const uint8_t *>(mCredentials[adjustedUserIndex].Get()),
+                                     sizeof(DlCredential) * mMaxCredentialsPerUser);
```

Comment:
This is writing out _one user's credentials_, no?  It's writing a buffer of size `sizeof(DlCredential) * mMaxCredentialsPerUser)`.

The corresponding read later will read `sizeof(DlCredential) * mMaxUsers * mMaxCredentialsPerUser` bytes and will read into a totally different place in memory.

Can we please stop the guess-and-check, and in the next update respond to this comment with a clear description of how storage is meant to work that I can then compare to the code?

Reply:
Sorry, I referred to https://github.com/project-chip/connectedhomeip/pull/20044 and updated a new commit.
It should write out all user's credentials.